website/addons/wiki/static/wikiPage.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+var ko = require('knockout');
+var $ = require('jquery');
+var $osf = require('osfHelpers');
+
+var mathrender = require('mathrender');
+var md = require('markdown').full;
+var mdQuick = require('markdown').quick;
+var diffTool = require('wikiDiff');
+
+var THROTTLE = 500;
+
+//<div id="preview" data-bind="mathjaxify">
+ko.bindingHandlers.mathjaxify = {
+    update: function(element, valueAccessor, allBindingsAccessor, data, context) {
+        var vm = context.$data;
+        //Need to unwrap the data in order for KO to know it's changed.
+        ko.unwrap(valueAccessor());
+
+        if(vm.allowMathjaxification() && vm.allowFullRender()) {
+            mathrender.mathjaxify('#' + element.id);
+        }
+    }
+};
+
+
+function ViewWidget(visible, version, viewText, rendered, contentURL, allowMathjaxification, allowFullRender, editor) {
+    var self = this;
+    self.version = version;
+    self.viewText = viewText; // comes from EditWidget.viewText
+    self.rendered = rendered;
+    self.visible = visible;
+    self.allowMathjaxification = allowMathjaxification;
+    self.editor = editor;
+    self.allowFullRender = allowFullRender;
+    self.renderTimeout = null;
+    self.displaySource = ko.observable('');
```

Comment:
displayText is a computedObservable but needs to do ajax. To get around the limitations of that, I use displaySource as the observable that will be used in later computedObservables.